catalog/tests/utilities/media_props_gen/test_column_parser.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from catalog.utilities.media_props_gen.helpers.column_parser import (
+    COLUMN_DEFINITIONS,
+    Column,
+)
+
+
+@pytest.mark.parametrize(
+    "init_args,expected_nullable,expected_str",
+    [
+        pytest.param(
+            {"required": True, "python_type": "URLColumn"},
+            False,
+            f"{COLUMN_DEFINITIONS['URLColumn']} (`upsert_strategy=newest_non_null, nullable=False, required=True`)",
+            id="required=True sets nullable to False",
+        ),
+        pytest.param(
+            {"required": False, "python_type": "StringColumn"},
+            True,
+            f"{COLUMN_DEFINITIONS['StringColumn']} (`upsert_strategy=newest_non_null, nullable=True, required=False`)",
+            id="required=False sets nullable to True",
+        ),
+        pytest.param(
+            {"name": "id", "db_name": "user_id", "python_type": "IntegerColumn"},
+            True,
+            f'{COLUMN_DEFINITIONS["IntegerColumn"]} '
+            f'(`name="id", upsert_strategy=newest_non_null, nullable=True, required=False`)',
+            id="name and db_name are different",
+        ),
+        pytest.param(
+            {"name": "id", "python_type": "StringColumn"},
+            True,
+            f"{COLUMN_DEFINITIONS['StringColumn']} (`upsert_strategy=newest_non_null, nullable=True, required=False`)",
+            id="default upsert strategy",
+        ),
+        pytest.param(
+            {"name": "genres", "required": False, "base_column": "StringColumn"},
+            True,
+            f"{COLUMN_DEFINITIONS['ArrayColumn']} "
+            f'(`name="genres", upsert_strategy=newest_non_null, nullable=True, required=False`)',
+            id="base_column",
+            marks=pytest.mark.xfail(reason="base_column is not implemented"),
```

Comment:
I think we want to use `pytest.mark.raises` here instead (since we can assert that the test raises a specific exception). Using `xfail` here will mean that _any_ failure, even an unrelated one, will "pass" this test.